src/engine/GroupByHashMapOptimization.h

```diff
@@ -8,25 +8,33 @@
 #include "engine/sparqlExpressions/RelationalExpressionHelpers.h"
 #include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
 
+// _____________________________________________________________________________
+static constexpr auto numericValueAdder =
+    []<typename T>(T value, double& sum, [[maybe_unused]] bool& error)
+        requires std::is_arithmetic_v<T> {
+  sum += static_cast<double>(value);
+};
+static constexpr auto nonNumericValueAdder =
+    [](sparqlExpression::detail::NotNumeric, [[maybe_unused]] double& sum,
+       bool& error) { error = true; };
+static constexpr auto valueAdder =
+    ad_utility::OverloadCallOperator(numericValueAdder, nonNumericValueAdder);
```

Comment:
Add a short comment and hide the two implementation lambdas from the outside world using an immediately invoked lambda:

static constexpr auto valueAdder = []() {
  // numericValueAdder
  // nonNumericValueAdder
   return OverloadCallOperator(...);
}(); // immediately call the lambda, so `valueAdder` has exactly the same content as before, but the two inner lambdas don't leak into the outer namespace.